src/stores/store/keys/local/local.go

```diff
@@ -0,0 +1,370 @@
+	key, err := s.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	secret, err := s.secretStore.Get(ctx, id, "")
+	if err != nil {
+		return nil, err
+	}
+
+	privkey, err := base64.StdEncoding.DecodeString(secret.Value)
+	if err != nil {
+		errMessage := "failed to decode private key secret"
+		logger.Error(errMessage)
+		return nil, errors.DependencyFailureError(errMessage)
+	}
+
+	switch {
+	case key.Algo.Type == entities.Eddsa && key.Algo.EllipticCurve == entities.Bn254:
+		return s.signEDDSA(privkey, data)
+	case key.Algo.Type == entities.Ecdsa && key.Algo.EllipticCurve == entities.Secp256k1:
+		return s.signECDSA(privkey, data)
+	default:
+		errMessage := "signing algorithm and curve combination not supported for signing"
+		logger.With("algorithm", key.Algo.Type, "curve", key.Algo.EllipticCurve).Error(errMessage)
+		return nil, errors.InvalidParameterError(errMessage)
+	}
+}
+
+func (s *Store) Verify(_ context.Context, pubKey, data, sig []byte, algo *entities.Algorithm) error {
+	return keys.VerifySignature(s.logger, pubKey, data, sig, algo)
+}
+
+func (s *Store) Encrypt(_ context.Context, id string, data []byte) ([]byte, error) {
+	return nil, errors.ErrNotImplemented
+}
+
+func (s *Store) Decrypt(_ context.Context, id string, data []byte) ([]byte, error) {
+	return nil, errors.ErrNotImplemented
+}
+
+func (s *Store) createKey(ctx context.Context, id string, importedPrivKey []byte, alg *entities.Algorithm, attr *entities.Attributes) (*entities.Key, error) {
+	logger := s.logger.With("id", id).With("algorithm", alg.Type).With("curve", alg.EllipticCurve)
+	logger.Debug("creating key")
+
+	var privKey []byte
+	var pubKey []byte
+	switch {
+	case alg.Type == entities.Eddsa && alg.EllipticCurve == entities.Bn254:
+		eddsaKey, err := eddsaBN254(importedPrivKey)
+		if err != nil {
+			errMessage := "failed to generate EDDSA/BN254 key pair"
+			logger.With("error", err).Error(errMessage)
+			return nil, errors.InvalidParameterError(errMessage)
+		}
+
+		privKey = eddsaKey.Bytes()
+		pubKey = eddsaKey.Public().Bytes()
+	case alg.Type == entities.Ecdsa && alg.EllipticCurve == entities.Secp256k1:
+		ecdsaKey, err := ecdsaSecp256k1(importedPrivKey)
+		if err != nil {
+			errMessage := "failed to generate Secp256k1/ECDSA key pair"
+			logger.With("error", err).Error(errMessage)
+			return nil, errors.InvalidParameterError(errMessage)
+		}
+
+		privKey = crypto.FromECDSA(ecdsaKey)
+		pubKey = crypto.FromECDSAPub(&ecdsaKey.PublicKey)
+	default:
+		errMessage := "invalid signing algorithm/elliptic curve combination"
+		logger.Error(errMessage)
+		return nil, errors.InvalidParameterError(errMessage)
+	}
+
+	secret, err := s.secretStore.Set(ctx, id, base64.StdEncoding.EncodeToString(privKey), attr)
+	if err != nil {
+		return nil, err
+	}
+
+	key := &entities.Key{
+		ID:        id,
+		PublicKey: pubKey,
+		Algo:      alg,
+		Metadata:  secret.Metadata,
+		Tags:      attr.Tags,
+	}
+
+	err = s.keysDA.Add(ctx, key)
+	if err != nil {
+		return nil, err
```

Comment:
if we fail to add it into DB we might have inconsistent state where the key is added into the SecretStore but not to the DB. I would rather implement a rollback strategy here

Reply:
added db tx